.github/workflows/doc-build.yml

```diff
@@ -9,140 +9,161 @@ on:
       - name: Merge CI fixes from sagemath/sage
         run: |
           mkdir -p upstream
           .ci/merge-fixes.sh 2>&1 | tee upstream/ci_fixes.log
         env:
           GH_TOKEN: ${{ github.token }}
           SAGE_CI_FIXES_FROM_REPOSITORIES: ${{ vars.SAGE_CI_FIXES_FROM_REPOSITORIES }}
-      - name: Store CI fixes in upstream artifact
+
+      # Building
+
+      - name: Generate Dockerfile
+        # From docker.yml
         run: |
-          if git format-patch --stdout test_base > ci_fixes.patch; then
-            cp ci_fixes.patch upstream/
-          fi
-      - uses: actions/upload-artifact@v3
-        with:
-          path: upstream
-          name: upstream
+          tox -e ${{ env.TOX_ENV }}
+          cp .tox/${{ env.TOX_ENV }}/Dockerfile .
+        env:
+          # Only generate the Dockerfile, do not run 'docker build' here
+          DOCKER_TARGETS: ""
 
-  build-docs:
-    runs-on: ubuntu-latest
-    container: ghcr.io/sagemath/sage/sage-ubuntu-focal-standard-with-targets:dev
-    needs: [get_ci_fixes]
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v4
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          driver-opts: network=host
 
-      - name: Update system packages
+      - name: Build Docker image
+        id: image
+        uses: docker/build-push-action@v5
+        with:
+          # push and load may not be set together at the moment
+          push:       true
+          load:       false
+          context:    .
+          tags:       ${{ env.BUILD_IMAGE }}
+          target:     with-targets
+          cache-from: type=gha
+          cache-to:   type=gha,mode=max
+          build-args: |
+            NUMPROC=6
+            USE_MAKEFLAGS=-k V=0 SAGE_NUM_THREADS=4 --output-sync=recurse
+            TARGETS_PRE=build/make/Makefile
+            TARGETS=ci-build-with-fallback
+
+      - name: Start container
         run: |
-          apt-get update && apt-get install -y git zip
+          docker run --name BUILD -dit \
+                     --mount type=bind,src=$(pwd),dst=$(pwd) \
+                     --workdir $(pwd) \
+                     ${{ env.BUILD_IMAGE }} /bin/sh
+
+      # Docs
 
-      - name: Add prebuilt tree as a worktree
+      - name: Store old docs
         id: worktree
         run: |
           git config --global --add safe.directory $(pwd)
           git config --global user.email "ci-sage@example.com"
-          git config --global user.name "Build & Test workflow"
-          # mathjax path in old doc
-          mathjax_path_from=$(SAGE_USE_CDNS=no /sage/sage -python -c "from sage_docbuild.conf import mathjax_path; print(mathjax_path)")
-          .ci/retrofit-worktree.sh worktree-image /sage
+          git config --global user.name "Build documentation workflow"
+          # mathjax path in old doc (regex)
+          mathjax_path_from="[-./A-Za-z_]*/tex-chtml[.]js"
           # mathjax path in new doc
```

Comment:
The original way to get `mathjax_path_from` seems safer.

Reply:
Perhaps but I would have to start the old container just for that...

Reply:
No. This
```
 # mathjax path in old doc
          mathjax_path_from=$(SAGE_USE_CDNS=no /sage/sage -python -c "from sage_docbuild.conf import mathjax_path; print(mathjax_path)")
```
just runs the same sage but with `SAGE_USE_CDNS=no`. You may run the same container for `mathjax_path_to`.

Reply:
Yes, this would take a couple of seconds more though.

Reply:
OK for now.